prow/cmd/branchprotector/protect.go

```diff
@@ -337,7 +345,12 @@ func (p *protector) UpdateRepo(orgName string, repoName string, repo config.Repo
 		}
 		for _, b := range bs {
 			_, ok := repo.Branches[b.Name]
-			if !ok && branchExclusions != nil && branchExclusions.MatchString(b.Name) {
+			if !ok && branchInclusions != nil && branchInclusions.MatchString(b.Name) {
+				branches[b.Name] = b
+			} else if !ok && branchInclusions != nil && !branchInclusions.MatchString(b.Name) {
+				logrus.Infof("%s/%s=%s: excluded", orgName, repoName, b.Name)
+				continue
+			} else if !ok && branchInclusions == nil && branchExclusions != nil && branchExclusions.MatchString(b.Name) {
```

Comment:
We know for sure at this point that `branchInclusions` is nil

```suggestion
			} else if !ok && branchExclusions != nil && branchExclusions.MatchString(b.Name) {
```